afl/src/lib.rs

```diff
@@ -18,6 +19,206 @@ unsafe extern "C" {
     static __afl_fuzz_ptr: *const u8;
 }
 
+// AFL++ IJON functions in afl-compiler-rt
+unsafe extern "C" {
+    pub fn ijon_max(addr: u32, val: u64);
+    pub fn ijon_min(addr: u32, val: u64);
+    pub fn ijon_set(addr: u32, val: u32);
+    pub fn ijon_inc(addr: u32, val: u32);
+    pub fn ijon_xor_state(val: u32);
+    pub fn ijon_reset_state();
+    pub fn ijon_simple_hash(x: u64) -> u64;
+    pub fn ijon_hashint(old: u32, val: u32) -> u32;
+    pub fn ijon_hashstr(old: u32, val: *const c_char) -> u32;
+    pub fn ijon_hashmen(old: u32, val: *const u8, len: usize) -> u32;
+    pub fn ijon_hashstack_backtrace() -> u32;
+    pub fn ijon_hashstack() -> u32;
+    pub fn ijon_strdist(a: *const u8, b: *const u8) -> u32;
+    pub fn ijon_memdist(a: *const u8, b: *const u8, len: usize) -> u32;
+    pub fn ijon_max_variadic(addr: u32, ...);
+    pub fn ijon_min_variadic(addr: u32, ...);
+}
+
+#[macro_export]
+macro_rules! ijon_inc {
+    ($x:expr) => {{
+        use std::sync::atomic::{AtomicU32, Ordering};
+        static LOC_CACHE: AtomicU32 = AtomicU32::new(0);
+        let mut loc = LOC_CACHE.load(Ordering::Relaxed);
+        if loc == 0 {
+            let cfile = CString::new(file!()).unwrap();
```

Comment:
```suggestion
            let cfile = std::ffi::CString::new(file!()).unwrap();
```
It is considered good practice to fully qualify identifiers in macros. That way, the macro importer doesn't have to import those identifiers as well, and the macro doesn't accidentally use something it didn't mean to.

So please make this change here and everywhere else `CString` is used in a macro.